cmd/release-controller/http.go

```diff
@@ -710,7 +710,48 @@ func (c *Controller) httpReleaseInfo(w http.ResponseWriter, req *http.Request) {
 
 	renderVerifyLinks(w, *info.Tag, info.Release)
 
-	if upgradesTo := c.graph.UpgradesTo(tag); len(upgradesTo) > 0 {
+	upgradesTo := c.graph.UpgradesTo(tag)
+
+	type releaseInfoStub struct {
+		Metadata   *struct {
+			Version  string   `json:"version"`
+			Previous []string `json:"previous"`
+		}               `json:"metadata"`
+	}
+	tagReleaseJson, err := c.releaseInfo.ReleaseInfo(tagPull)
+	var requiredUpgrades []string
+	var tagReleaseInfo releaseInfoStub
+	var missingUpgrades int
+	err = json.Unmarshal([]byte(tagReleaseJson), &tagReleaseInfo)
```

Comment:
This feels super ugly - you should be using a wrapper, not parsing it yourself.

The terminology is wierd here - these aren't requiredUpgrades, they are explicit or supported upgrades (they are explicitly defined in the release, or they are describe as supported by the release)